### PR TITLE
Ensure that the waitForBody yields at the earliest possible time when body is opened

### PR DESCRIPTION
**Pull request by dvoytenko**
TODO:

- [x] Submit fix all new uses of waitForBody where the complete body is expected (#22344).
- [x] Rename `waitForBody` to `waitForBodyOpen` (#22347).
- [x] Submit this PR.
- [ ] Exhale.

### CODE REVIEW

File: test/unit/test-element-service.js [hunk 1]
```diff
@@ -38,30 +38,11 @@ import {
 describe('getElementServiceIfAvailable()', () => {
   let doc;
   let win;
-  let readyStateListeners;
```
Comment (jridgewell) [inline test/unit/test-element-service.js]: I'm assuming this is just a revert of the original PR?

File: test/unit/test-dom.js [hunk 2]
```diff
@@ -747,6 +747,38 @@ describes.sandboxed('DOM', {}, env => {
...
+
+    it('should yield body asap even if doc is not complete', () => {
+      return new Promise((resolve, reject) => {
+        const doc = {
+          readyState: 'loading',
+          body: null,
+          documentElement: {
+            ownerDocument: {
+              defaultView: {
+                setInterval() {
```
Comment (jridgewell) [inline test/unit/test-dom.js]: Could just `setInterval: window.setInterval`?

File: test/unit/test-element-service.js (hunk 1 above)
Comment (dvoytenko) [reply test/unit/test-element-service.js]: Pretty much, yes.

File: test/unit/test-dom.js (hunk 2 above)
Comment (dvoytenko) [reply test/unit/test-dom.js]: I tried. It gets "invalid invocation". I guess it gets the wrong `this`.

**Pull request merged**